### PR TITLE
Add rspec tests for the extension file

### PR DESCRIPTION
**Pull request by josephwegner**
## What does this PR do?

Adds a structure for rspec unit tests, as well as some very basic tests for the main extension file
## How should this be tested?

Pull down the PR and:

``` sh
bundle install
bundle exec rspec
```

That should test both that I actually _did_ set up rspec, and also that my tests pass :)

### CODE REVIEW

File: spec/support/fixture.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+rescue LoadError
+end
+
+
+module Middleman
+  module Fixture
+
+    class << self
+
+      def app &block
```
Comment (joannecheng) [inline spec/support/fixture.rb]: Parenthesis around arguments in function definitions (https://github.com/bbatsov/ruby-style-guide#syntax)

File: spec/support/given.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+      `rsync -av ./spec/fixtures/#{fixture}/ #{TMP}/`
+      Dir.chdir TMP
+    end
+
+    def file name, content
+      file_path = File.join( TMP, name )
+      FileUtils.mkdir_p( File.dirname(file_path) )
+      File.open( file_path, 'w' ) do |file|
+        file.write content
```
Comment (joannecheng) [inline spec/support/given.rb]: parenthesis around arguments

File: spec/support/given.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      end
+    end
+
+    def cleanup!
+      Dir.chdir ROOT
```
Comment (joannecheng) [inline spec/support/given.rb]: Parenthesis around arguments.

File: spec/support/fixture.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
+begin
+  require 'middleman-core/rack'
+rescue LoadError
```
Comment (joannecheng) [inline spec/support/fixture.rb]: is there any reason why we want to rescue a `LoadError`?

**Comment by josephwegner** [comment]
@joannecheng Heh - sry for all the styleguide breaks from the support files. I totally just stole those from an example middleman app :P

I'll fix them!

File: spec/support/fixture.rb (hunk 4 above)
Comment (josephwegner) [reply spec/support/fixture.rb]: I don't specifically know... I will remove it, and if it ever freaks out in the future we can readd it.

#ITotallyStoleAllThisCodeFromTheInternetYOLO

**Comment by josephwegner** [comment]
@joannecheng all cleaned up

**Comment by joannecheng** [comment]
Tests all passed! Thanks for cleaning it up 🚢

**Comment by josephwegner** [comment]
Dope!

**Pull request merged**